### PR TITLE
Added some extra logging to MetricsValidator module

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
This is a small PR that:
- adds some extra logging to MetricsValidator. This is needed to have better visibility into how long does `ValidateMetrics` take and if there are any module disconnects. 
- decreases Direct Method call timeout in `ValidateMetrics` test. Counterintuitively, I believe this should improve the stability of the test. Right now we use the same 5m timeout on the DM call as the whole test timeout (5m as well). Sometimes we observe that IoT Hub fails (or takes long time) to reply to Direct Method (Service API) call. Having 5m timeout on the call gives us only one chance. Reducing the timeout (to 180 sec) gives us the ability to retry one more time during the max duration of the test (5m).

These change do not require test coverage.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### CODE REVIEW

File: test/modules/MetricsValidator/src/Program.cs [hunk 1]
```diff
@@ -71,10 +71,16 @@ static async Task<int> MainAsync()
...
-                            return new MethodResponse(Encoding.UTF8.GetBytes(testReporter.ReportResults()), (int)HttpStatusCode.OK);
+                            var result = new MethodResponse(Encoding.UTF8.GetBytes(testReporter.ReportResults()), (int)HttpStatusCode.OK);
+
+                            Logger.LogInformation($"Finished validating metrics. Result size: {result.Result.Length}");
+                            return result;
                         },
                         null);
 
+                    moduleClient.SetConnectionStatusChangesHandler((status, reason)
+                        => Logger.LogWarning($"IoT Hub connection status Changed Status: {status} Reason: {reason}"));
```
Comment (and-rewsmith) [inline test/modules/MetricsValidator/src/Program.cs:82]: Won't this detect connection status changed with EH? And if Hub closes the connection doesn't EH try to recreate it without disconnecting the module and triggering this callback?

Comment (vadim-kovalyov) [reply test/modules/MetricsValidator/src/Program.cs:82]: Yes, that's true. But this is for unlikely case when the module itself looses the connection.

Comment (and-rewsmith) [reply test/modules/MetricsValidator/src/Program.cs:82]: I was trying to say that the warning is indicating a connection status with IoTHub, but this callback is for connection status changes with module->EH. Should we change the warning name?

**Review by and-rewsmith — DISMISSED** [review]

File: test/modules/MetricsValidator/src/Program.cs (hunk 1 above)
Comment (vadim-kovalyov) [reply test/modules/MetricsValidator/src/Program.cs]: Ah, I see. I've updated the message.

**Comment by vadim-kovalyov** [comment]
@and-rewsmith I've also made change to direct method timeouts as part of this PR. Hope this is fine. I can split it into a separate PR if needed.

**Review by and-rewsmith — DISMISSED** [review]

**Review by and-rewsmith — APPROVED** [review]

**Pull request merged**